pandas/tests/test_lib.py

```diff
@@ -100,9 +101,14 @@ def test_maybe_indices_to_slice_right_edge(self):
         assert not isinstance(maybe_slice, slice)
         tm.assert_numpy_array_equal(maybe_slice, indices)
 
-        with pytest.raises(IndexError):
+        msg = (
+            "index 100 is out of bounds for axis 0 with size 100|"
```

Comment:
minor but could simplify with a group or even character class

```suggestion
            "index 100 is out of bounds for axis (0|1) with size 100|"
```

Reply:
That's really smart!